ui/components/app/confirmations/alerts/alert-modal/alert-modal.tsx

```diff
@@ -40,12 +41,14 @@ export type AlertModalProps = {
    * It override `startAccessory` of ModalHeaderDefault and by default no content is present.
    */
   headerStartAccessory?: React.ReactNode;
-  /** The owner ID of the relevant alert from the `confirmAlerts` reducer. */
-  ownerId: string;
   /** The function invoked when the user acknowledges the alert. */
   onAcknowledgeClick: () => void;
+  /** The function to execute a determinate action based on the action key. */
+  onActionClick?: (actionKey: string) => void;
```

Comment:
As mentioned in the other comment, would it be cleaner if we remove this and instead use a React context so we don't have to pass around handlers to every row and then to the modal?